internal/tools/k8s/helpers.go

```diff
@@ -0,0 +1,33 @@
+package k8s
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// WaitForPodRunning will poll a given kubernetes pod at a regular interval for
+// it to enter the 'Running' state. If the pod fails to become ready before
+// the provided timeout then an error will be returned.
+func (kc *KubeClient) WaitForPodRunning(name, namespace string, timeout int) (apiv1.Pod, error) {
+	timer := time.NewTimer(time.Duration(timeout) * time.Second)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-timer.C:
+			return apiv1.Pod{}, errors.New("timed out waiting for pod to become ready")
+		default:
+			pods, _ := kc.GetPods(namespace)
+			for _, pod := range pods {
+				if strings.Contains(pod.Name, name) && pod.Status.Phase == apiv1.PodRunning {
```

Comment:
The `strings.Contains` seems fragile in the face of some kind of naming overlap. Can we pass in a more precise name, or compare against something else?

Reply:
You are right that there could be name overlap and that this is fragile to some degree.

Based on my understanding of kubernetes, I think that we will need to do sequential lookups based on the deployment of the pod (if the pod is created from a deployment). I will create a ticket to capture this for a future improvement.